core-java/src/test/java/com/baeldung/arrays/ArraysTest.java

```diff
@@ -0,0 +1,148 @@
+
+    @Test
+    public void testArraysCopyOf(){
+        String[] revised = Arrays.copyOf(intro, 3);
+        String[] expanded = Arrays.copyOf(intro, 5);
+
+        Assert.assertArrayEquals(Arrays.copyOfRange(intro, 0, 3), revised);
+        Assert.assertNull(expanded[4]);
+    }
+
+    @Test
+    public void testArraysFill(){
+        String[] stutter = new String[3];
+        Arrays.fill(stutter, "once");
+
+        Assert.assertTrue(Stream.of(stutter)
+                .allMatch(el -> "once".equals(el)));
+    }
+
+    @Test
+    public void testArraysEquals(){
+        Assert.assertTrue(Arrays.equals(new String[]{"once","upon","a","time"}, intro));
+        Assert.assertFalse(Arrays.equals(new String[]{"once","upon","a", null}, intro));
+    }
+
+    @Test
+    public void testArraysDeepEquals() {
+        String[] end = {"the", "end"};
+        Object[] story =
+                new Object[] { intro, new String[] { "chapter one", "chapter two" }, end };
+        Object[] copy =
+                new Object[] { intro, new String[] { "chapter one", "chapter two" }, end };
+
+        Assert.assertTrue(Arrays.deepEquals(story, copy));
+        Assert.assertFalse(Arrays.equals(story, copy));
+    }
+
+    @Test
+    public void testArraysSort(){
+        String[] sorted = Arrays.copyOf(intro, 4);
+        Arrays.sort(sorted);
+
+        Assert.assertArrayEquals(new String[]{"a", "once", "time", "upon"}, sorted);
+    }
+
+    @Test
+    public void testArraysBinarySearch(){
+        String[] sorted = Arrays.copyOf(intro, 4);
+        Arrays.sort(sorted);
+        int exact = Arrays.binarySearch(sorted, "time");
+        int caseInsensitive = Arrays.binarySearch(sorted, "TiMe", String::compareToIgnoreCase);
+
+        Assert.assertEquals("time", sorted[exact]);
+        Assert.assertEquals(2, exact);
+        Assert.assertEquals(exact, caseInsensitive);
+    }
+
+    @Test
+    public void testArraysHashCode(){
+        int beforeChange = Arrays.hashCode(intro);
+        intro[3] = null;
+        int afterChange = Arrays.hashCode(intro);
+
+        Assert.assertNotEquals(beforeChange, afterChange);
+    }
+
+    @Test
+    public void testArraysDeepHash(){
+        Object[] looping = new Object[]{ intro, intro };
+        int deepHashBefore = Arrays.deepHashCode(looping);
+        int hashBefore = Arrays.hashCode(looping);
+
+        intro[3] = null;
+
+        int hashAfter = Arrays.hashCode(looping);
+        int deepHashAfter = Arrays.deepHashCode(looping);
+
+        Assert.assertEquals(hashAfter, hashBefore);
+        Assert.assertNotEquals(deepHashAfter, deepHashBefore);
+    }
+
+    @Test
+    public void testStream(){
+        Assert.assertEquals(Arrays.stream(intro).count(), 4);
+
+        exception.expect(ArrayIndexOutOfBoundsException.class);
+        Arrays.stream(intro, 2, 1).count();
+    }
+
+    @Test
+    public void testSetAll(){
+        String[] longAgo = new String[4];
+        Arrays.setAll(longAgo, i -> this.getWord(i));
```

Comment:
Up to you, but you could simplify this by doing something like:

```java
Arrays.setAll(intro, i -> intro[i].toUpperCase())
```

And then just put the `intro` instantiation into a `@Before` method.

Up to you, though.